### PR TITLE
feat: support physical LoadDataPlan in spark planner

### LINKED ISSUE #726: Support physical plan of "LOAD DATA" in Spark planner
Support importing data with Spark in cluster mode.

### PR DESCRIPTION
**Pull request by vagetablechicken**
Closes #726 
Not full functioned, need to support to get offline address from nameserver. And no unit test about hdfs.

### CODE REVIEW

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    val inputFile = node.File()
+    val spark = ctx.getSparkSession
+
+    // read input file
+    val (format, options) = parseOptions(node)
+    logger.info("format {}, options {}", format, options: Any)
+    val df = spark.read.options(options).format(format).load(inputFile)
+
+    // write, offline address may contains some files, use append mode
+    df.write.mode("append").parquet(offlineAddress)
```
Comment (tobegit3hub) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: Choose `append` or `overwrite` from options. And default value should be `error_if_exists`.

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+object LoadDataPlan {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  def parseOptions(node: PhysicalLoadDataNode): (String, Map[String, String]) = {
+    var format = "csv"
+    var option = node.GetOption("format")
+    if (option != null) {
+      val f = option.GetStr()
+      if (f.equalsIgnoreCase("parquet")) {
```
Comment (tobegit3hub) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: We can use `match` syntax and switch-case in Scala.

Comment (vagetablechicken) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: ok

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala (hunk 1 above)
Comment (vagetablechicken) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: So the offline address which we get from nameserver should be a new path?
I don't think this is good that `LOAD DATA` has the option `(write)mode`. Too confused.

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala (hunk 2 above)
Comment (vagetablechicken) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: There's no need to match-case, options["format"] is a string, just need to get lowercase and assign it to return value.

**Comment by tobegit3hub** [comment]
CICD failed and make sure ` com.4paradigm.hybridse:hybridse-sdk:jar:0.4.0-SNAPSHOT`  has been deployed.

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala (hunk 1 above)
Comment (tobegit3hub) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: Users may `LOAD DATA` for multiple times and we may support different modes including `overwrite` and `append`.

Comment (vagetablechicken) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: but the offline address is a path, so we will create a new path for a file? e.g. load data infile f1.csv, we will get a offline address like 'f1_csv_path'?

Comment (tobegit3hub) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: Yes. For deep-copy mode(not symbolic mode), we will generate a new path(mostly HDFS path). Then load data and write into new path. Users may specify overwriting the old imported data or append to the old imported data.

Comment (vagetablechicken) [reply java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/LoadDataPlan.scala]: ok, so `LOAD DATA` will have the option `mode`. I'll fix it.

**Review by aceforeverd — APPROVED** [review]

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Comment by vagetablechicken** [comment]
[data import for offline and online in the task manager](https://trello.com/c/OqReyYcD/23-data-import-for-offline-and-online-in-the-task-manager)

**Pull request merged**